app/models/miq_server/queue_management.rb

```diff
@@ -11,6 +11,15 @@ def clear_miq_queue_for_this_server
     end
   end
 
+  def clean_stop_worker_queue_items
+    MiqQueue.where(
+      :class_name  => self.class.name,
+      :method_name => "stop_worker",
+      :queue_name  => 'miq_server',
+      :server_guid => guid
+    ).destroy_all
+  end
```

Comment:
Alternatively you can just leave these on the queue, and then just let it gracefully fail.  This way, if/when we move to kafka it will do the same thing, and we don't have to deal with introspecting the queue and deleting records, which is much harder with

Reply:
Sure? I was trying not to change any behavior but dropping this would make it simpler

We also have https://github.com/ManageIQ/manageiq/pull/21445/files#diff-2902f342d0ecf5ba28f9fb3a2f9b9ef9c6b7fd0989d0600f29c7a579917a14a3R4-R14 which indicates it is deprecated so I need to dig into if it is still used anywhere

Reply:
Oh that;s fine - if the goal is 1:1 parity, then yes I agree, and then we can do these kinds of changes in a followup - maybe open an issue to track that afterwards.